bittensor/_subtensor/subtensor_impl.py

```diff
@@ -1500,6 +1513,91 @@ def neurons(self, block: int = None ) -> List[SimpleNamespace]:
                 break
         return neurons
 
+    def neurons_fast(self, block: int = None ) -> List[SimpleNamespace]: 
+        r""" Returns a list of neuron from the chain, using the bundled subtensor-node-api
+        Args:
+            block (int):
+                block to sync from.
+        Returns:
+            neuron (List[SimpleNamespace]):
+                List of neuron objects.
+        """
+        neurons = []
+        if block is None:
+            block: int = self.get_current_block()
+
+        block_hash: str = self.substrate.get_block_hash( block )
+        endpoint_url: str = self.chain_endpoint
+        # format endpoint with ws://
+        if "ws://" not in endpoint_url and "wss://" not in endpoint_url:
+            endpoint_url = "ws://" + endpoint_url
+        
+        try:
+            if platform == "linux" or platform == "linux2":
+                # linux
+                os_name = "linux"
+            elif platform == "darwin":
+                # OS X
+                os_name = 'macos'
+            else:
+                raise Exception('Unsupported platform for subtensor-node-api')
+            bittensor.__console__.print("Using subtensor-node-api for neuron retrieval...")
+            path_to_bin = os.path.join(os.path.dirname(__file__), f"../../bin/subtensor-node-api-{os_name}")
+            # will write to ~/.bittensor/metagraph.json by default
+            subprocess.run([path_to_bin, "sync_and_save", "-u", endpoint_url, '-b', block_hash], check=True, stdout=subprocess.PIPE)
+            with open(os.path.join(os.path.expanduser('~'), '.bittensor/metagraph.json')) as f:
+                data = json.load(f)
+
+            # all the large ints are strings
+            RAOPERTAO = 1000000000
+            U64MAX = 18446744073709551615
+            """
+            We expect a JSON array of:
+            {
+                "uid": int,
+                "ip": str,
+                "ip_type": int,
+                "port": int,
+                "stake": str(int),
+                "rank": str(int),
+                "emission": str(int),
+                "incentive": str(int),
+                "consensus": str(int),
+                "trust": str(int),
+                "dividends": str(int),
+                "modality": int,
+                "last_update": str(int),
+                "version": int,
+                "priority": str(int),
+                "last_update": int,
+                "weights": [
+                    [int, int],
+                ],
+                "bonds": [
+                    [int, str(int)],
+                ],
+            }
+            """
```

Comment:
This is helpful given that you are including an external binary.

What about moving this specification to a document in the bin file and translate this comment also to a set of unit tests?

Reply:
With both thing i suggested i would reference both with links here

Reply:
Adding unit tests for the binary file is good to:
- localize files and ensure that you have all of them in the repo
- write a specification

Reply:
Agree, @camfairchild this should be moved to a proper documentation as well. Nicely done!